core-rust/state-manager/src/store/historical_state.rs

```diff
@@ -237,54 +263,48 @@ impl<'s, R: ReadableRocks> VersionScopedSubstateDatabase<'s, StateManagerDatabas
     /// Returns the state version at which this store is scoped.
     pub fn at_state_version(&self) -> StateVersion {
         match self {
-            VersionScopedSubstateDatabase::Current {
+            VersionScopedDatabase::Current {
                 current_version, ..
             } => *current_version,
-            VersionScopedSubstateDatabase::Historical(database) => database.at_state_version(),
+            VersionScopedDatabase::Historical(database) => database.at_state_version,
         }
     }
 
-    fn current(database: &'s StateManagerDatabase<R>, current_version: StateVersion) -> Self {
+    /// Returns the header of the proof proving the scoped point on the ledger.
+    pub fn proving_ledger_header(&self) -> LedgerHeader {
```

Comment:
A bit of naming nitpicking: this is really the _latest_ ~proof~ header, as seen by the current scope, so maybe just: `latest_ledger_header`? One might argue that this can be a bit ambiguous when reading the code (because one has to be aware of the fact that the scoped db is being used), but the same argument could apply to `get_substate` - yet we haven't introduced a separate `get_scoped_substate` (and I'm not suggesting that we do).

Reply:
ah, nvm, this can also be a future proof (from the perspective of the scoped db)

Reply:
`proving_ledger_header` is fine then :+1:

Reply:
For the record, as discussed offline, it might turn out that this will no longer be needed in its current form.